fdbclient/include/fdbclient/CoordinationInterface.h

```diff
@@ -350,4 +350,6 @@ struct CheckDescriptorMutableRequest {
 	}
 };
 
+IPAddress determinePublicIPAutomatically(ClusterConnectionString& ccs);
```

Comment:
I don't like that we put declarations in random header files all over the place. This function is defined in `AutoPublicAddress.cpp`. So the declaration should go into `AutoPublicAddress.h` (which needs to be created first).

Reply:
I agree this is not in its final state, I was trying to minimize the change set because this needs to be cherry-picked to two release branches and we changed our header file organization a while back.

Also, the final state should be different than what you suggest.   I do not see the point in creating a separate header file for this.   `fdbclient/AutoPublicAddress.cpp` contains a single function, and its argument type is defined in `fdbclient/CoordinationInterface.h`, so I do not see why this is a separate file to begin with.  `CoordinationInterface.cpp` does not exist yet, but that seems like a more sensible place for it.

This function should maybe even be a member function of the `ClusterConnectionString` class since that's its only argument.